orttraining/orttraining/training_ops/cpu/nn/batch_norm_grad.cc

```diff
@@ -0,0 +1,109 @@
+/**
+* Copyright (c) 2016-present, Facebook, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/* Modifications Copyright (c) Microsoft. */
+
+#include "batch_norm_grad.h"
+#include "core/util/math.h"
+#include "core/util/math_cpuonly.h"
+#include "core/providers/common.h"
+#include "core/framework/op_kernel_context_internal.h"
+#include "core/common/safeint.h"
+#include "core/providers/cpu/nn/batch_norm_helper.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+template <typename T>
+Status BatchNormalizationGrad<T>::Compute(OpKernelContext* ctx) const {
+  const Tensor* dY = ctx->Input<Tensor>(0);
+  const Tensor* X = ctx->Input<Tensor>(1);
+  const Tensor* Scale = ctx->Input<Tensor>(2);
+  const Tensor* saved_mean = ctx->Input<Tensor>(3);
+  const Tensor* saved_inv_variance = ctx->Input<Tensor>(4);
```

Comment:
Just noticed a minor thing on BatchNormInternal's schema, 
.Output(4, "saved_inv_std", "Inverse standard deviation for the batch", "U", OpSchema::Optional, true, 1, OpSchema::NonDifferentiable) 


Output4's name should actually be save_inv_variance.